### PR TITLE
feat: change the output Image Stream name and tag

### LINKED ISSUE #337: Enhancement: ability to specify output image name
It would be nice to have the ability to give a name to the output image from the `build` task. It's nice that nodeshift already uses the `name` from `package.json`. But it would be good for pipelines to be able to specify a given run what the output image name - for example `my-webapp-staging` and `my-webapp-production`.

### PR DESCRIPTION
**Pull request by lholmquist**
fixes #337

@lance when you get a chance,  you want to test this out.

I still need to added tests and update the readme and and stuff.  but the functionality should all be there

there are now  `--outputImageStream` and `--outputImageStreamTag`  flags to specify the different output stream names and tag

### CODE REVIEW

**Comment by lance** [comment]
Will do perhaps tomorrow or early next week. Feel free to bug me about it
again if it slips through the cracks.

File: README.md [hunk 1]
```diff
@@ -155,6 +155,12 @@ Specify the tag of the docker image to use for the deployed application. default
 #### dockerImage
 Specify the s2i builder image of Node.js to use for the deployed applications.  Defaults to [nodeshift/centos7-s2i-nodejs](https://hub.docker.com/r/nodeshift/centos7-s2i-nodejs)
 
+#### outputImageStream
+The name of the Image Stream to output to.  Defualts to project name from package.json
```
Comment (helio-frota) [inline README.md]: Minor typo

File: index.js [hunk 2]
```diff
@@ -18,6 +18,8 @@ const cli = require('./bin/cli');
   @param {boolean} [options.namespace.create] - flag to create the namespace if it does not exist. Only applicable for the build and deploy command. Must be used with namespace.name
   @param {string} [options.namespace.name] - flag to specify the project namespace name to build/deploy into.  Overwrites any namespace settings in your OpenShift or Kubernetes configuration files
   @param {string} [options.imageTag] - set the version to use for the nodeshift/centos7-s2i-image.  Versions are docker hub tags: https://hub.docker.com/r/nodeshift/centos7-s2i-nodejs/tags/
+  @param {string} [options.outputImageStream] - the name of the Image Stream to output to.  Defualts to project name from package.json
```
Comment (helio-frota) [inline index.js]: Minor typo `Defualts` maybe copy/past from previous

Comment (lholmquist) [reply index.js]: yea,  probably same for all the copy/pastes in that file too :)

**Comment by lholmquist** [comment]
@lance just a friendly ping :)

File: bin/nodeshift [hunk 3]
```diff
@@ -53,6 +53,15 @@ yargs
     type: 'string',
     default: 'latest'
   })
+  .options('outputImageStream', {
+    describe: 'The name of the Image Stream to output to.  Defaults to project name from package.json',
+    type: 'string'
+  })
+  .options('outputImageStreamTag', {
+    describe: 'The Tag of the Image Stream to output to.',
```
Comment (lance) [inline bin/nodeshift]: Minor nit: you don't need to capitalize `Tag` here. Also, here and other places, if you are going to capitalize "Image Stream", you might as well write it as the kube resource `ImageStream`.

File: lib/build-config.js [hunk 4]
```diff
@@ -78,8 +78,7 @@ async function removeBuildsAndBuildConfig (config) {
 async function createOrUpdateBuildConfig (config) {
   // Check for a BuildConfig, create or update if necessary
   const buildName = config.buildName;
-  const imageStreamName = config.projectName;
-  const outputImageStreamTag = `${imageStreamName}:latest`; // TODO: base on a tag
+  const outputImageStreamTag = `${config.outputImageStreamName}:latest`; // TODO: base on a tag
```
Comment (lance) [inline lib/build-config.js]: Shouldn't you be using the `outputImageSreamTag` option here instead of `latest`?

File: lib/nodeshift-config.js [hunk 5]
```diff
@@ -60,6 +60,8 @@ async function setup (options = {}) {
     throw new Error('"name" in package.json can only consist lower-case letters, numbers, and dashes. It must start with a letter and can\'t end with a -.');
   }
 
+  options.outputImageStreamName = options.outputImageStreamName || projectPackage.name;
+  options.outputImageStreamTag = 'latest';
```
Comment (lance) [inline lib/nodeshift-config.js]: Shouldn't this be `options.outputImageStreamTag || `latest`?

File: lib/build-config.js (hunk 4 above)
Comment (lholmquist) [reply lib/build-config.js]: yea,  probably :)

File: lib/nodeshift-config.js (hunk 5 above)
Comment (lholmquist) [reply lib/nodeshift-config.js]: same answer as above

Comment (lholmquist) [reply lib/nodeshift-config.js]: looks like i forgot to write a test that would have caught that

File: lib/build-config.js (hunk 4 above)
Comment (lholmquist) [reply lib/build-config.js]: Updated with the new commits

File: lib/nodeshift-config.js (hunk 5 above)
Comment (lholmquist) [reply lib/nodeshift-config.js]: updated with the new commits

**Pull request merged**